projects/storefrontlib/src/cms-components/order-confirmation/components/order-confirmation-overview/order-confirmation-overview.component.ts

```diff
@@ -37,17 +37,164 @@ export class OrderConfirmationOverviewComponent implements OnInit, OnDestroy {
+        map((textTitle) => {
+          const date = this.getDate(new Date(isoDate));
+
+          return {
+            title: textTitle,
+            text: [date],
+          };
+        })
+      );
+  }
+
+  getOrderCodeCardContent(orderCode: string): Observable<Card> {
+    return this.translation
+      .translate('checkoutOrderConfirmation.orderNumber')
+      .pipe(
+        filter(() => Boolean(orderCode)),
+        map((textTitle) => ({
+          title: textTitle,
+          text: [orderCode],
+        }))
+      );
+  }
+
+  getOrderStatusCardContent(status: string): Observable<Card> {
+    return combineLatest([
+      this.translation.translate('checkoutOrderConfirmation.status'),
+      this.translation.translate('orderDetails.statusDisplay', {
+        context: status,
+      }),
+    ]).pipe(
+      map(([textTitle, textStatus]) => ({
+        title: textTitle,
+        text: [textStatus],
+      }))
+    );
+  }
+
+  getPurchaseOrderNumber(poNumber: string): Observable<Card> {
+    return combineLatest([
+      this.translation.translate('checkoutReview.poNumber'),
+      this.translation.translate('checkoutPO.noPoNumber'),
+    ]).pipe(
+      map(([textTitle, noneTextTitle]) => ({
+        title: textTitle,
+        text: [poNumber ? poNumber : noneTextTitle],
+      }))
+    );
+  }
+
+  getMethodOfPaymentCardContent(
+    hasPaymentInfo: PaymentDetails
+  ): Observable<Card> {
+    return combineLatest([
+      this.translation.translate('checkoutProgress.methodOfPayment'),
+      this.translation.translate('paymentTypes.paymentType_ACCOUNT'),
+      this.translation.translate('paymentTypes.paymentType_CARD'),
+    ]).pipe(
+      map(([textTitle, textAccount, textCard]) => ({
+        title: textTitle,
+        text: [Boolean(hasPaymentInfo) ? textCard : textAccount],
+      }))
+    );
+  }
+
+  getCostCenterCardContent(costCenter: CostCenter): Observable<Card> {
+    return this.translation.translate('checkoutPO.costCenter').pipe(
+      filter(() => Boolean(costCenter)),
+      map((textTitle) => ({
+        title: textTitle,
+        textBold: costCenter?.name,
+        text: ['(' + costCenter?.unit.name + ')'],
+      }))
+    );
+  }
+
   getAddressCardContent(deliveryAddress: Address): Observable<Card> {
     return this.translation.translate('addressCard.shipTo').pipe(
       filter(() => Boolean(deliveryAddress)),
       map((textTitle) => ({
         title: textTitle,
         textBold: `${deliveryAddress.firstName} ${deliveryAddress.lastName}`,
         text: [
-          deliveryAddress.line1,
-          deliveryAddress.line2,
-          `${deliveryAddress.town}, ${deliveryAddress.country.isocode}, ${deliveryAddress.postalCode}`,
-          deliveryAddress.phone,
+          `${deliveryAddress.line1}, ${deliveryAddress.town}, ${deliveryAddress.country.isocode}`,
+          deliveryAddress.line2
+            ? `${deliveryAddress.line2}, ${deliveryAddress.town}, ${deliveryAddress.country.isocode}`
```

Comment:
Why would you display the `town` and `country` with the `line2` if it's already displayed with the `line1`?

Reply:
Should I just display line 2 alone instead, I was thinking it would be weird, but now that you mention it, I believe it's weird right now too haha. Made it like that for consistency, but I guess it's weird right now like this

Reply:
Is there a mock?

Reply:
for that specific case, no :P 

Truthy mocks was the wires sent to us, but this specific case for overview was overwritten by the mocks given by Bill.

So there's no source of truth for this case! I'll follow up with Bill on what to do for it

Reply:
using formattedAddress and customer will take it from there according to Bill. This change was done in b2bcheckout branch